src/Chemistry.hpp

```diff
@@ -77,11 +83,19 @@ template <typename problem_t> void computeChemistry(amrex::MultiFab &mf, const R
 			// call the EOS to set initial internal energy e
 			eos(eos_input_re, chemstate);
 
+			if (dt < 0) {
+				amrex::Abort("Cannot do chemistry with dt < 0!");
```

Comment:
Can you move this check outside of the inner loop? This has severe performance implications on GPU: https://quokka-astro.github.io/quokka/error_checking.html#abort and https://quokka-astro.github.io/quokka/performance.html#gpu-hardware-characteristics